From 85dae6b26be348ecb8fbba2e9e1587766eb01ef8 Mon Sep 17 00:00:00 2001
From: Dmitry Kasatkin <d.kasatkin@samsung.com>
Date: Fri, 8 Nov 2013 16:09:06 +0200
Subject: [PATCH 30/45] ima: export ima_fix_xattr

Export ima_fix_xattr() to be used by directory verification code.

Signed-off-by: Dmitry Kasatkin <d.kasatkin@samsung.com>
---
 security/integrity/ima/ima.h          | 1 +
 security/integrity/ima/ima_appraise.c | 3 +--
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/security/integrity/ima/ima.h b/security/integrity/ima/ima.h
index 22ca359..55a2448 100644
--- a/security/integrity/ima/ima.h
+++ b/security/integrity/ima/ima.h
@@ -223,6 +223,7 @@ enum hash_algo ima_get_hash_algo(struct evm_ima_xattr_data *xattr_value,
 				 int xattr_len);
 int ima_read_xattr(struct dentry *dentry,
 		   struct evm_ima_xattr_data **xattr_value);
+int ima_fix_xattr(struct dentry *dentry, struct integrity_iint_cache *iint);
 
 #else
 static inline int ima_appraise_measurement(int func,
diff --git a/security/integrity/ima/ima_appraise.c b/security/integrity/ima/ima_appraise.c
index a0aa32c..bdbb728 100644
--- a/security/integrity/ima/ima_appraise.c
+++ b/security/integrity/ima/ima_appraise.c
@@ -44,8 +44,7 @@ int ima_must_appraise(struct dentry *dentry, int mask, enum ima_hooks func)
 	return ima_match_policy(dentry, func, mask, IMA_APPRAISE, NULL);
 }
 
-static int ima_fix_xattr(struct dentry *dentry,
-			 struct integrity_iint_cache *iint)
+int ima_fix_xattr(struct dentry *dentry, struct integrity_iint_cache *iint)
 {
 	int rc, offset;
 	u8 algo = iint->ima_hash->algo;
-- 
2.1.4

